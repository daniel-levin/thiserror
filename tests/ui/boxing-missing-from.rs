use thiserror::Error;

#[derive(Error, Debug)]
pub enum Missing {
    #[error(transparent)]
    NotThere(#[boxing] std::io::Error),
}

#[derive(Error, Debug)]
#[error("...")]
pub struct AlsoMissing {
    #[boxing] err: std::io::Error,
}

fn main() {}
