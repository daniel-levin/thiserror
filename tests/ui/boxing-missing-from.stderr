error: attribute #[boxing] requires the field to also have a #[from] attribute
 --> tests/ui/boxing-missing-from.rs:6:14
  |
6 |     NotThere(#[boxing] std::io::Error),
  |              ^^^^^^^^^

error: attribute #[boxing] requires the field to also have a #[from] attribute
  --> tests/ui/boxing-missing-from.rs:12:5
   |
12 |     #[boxing] err: std::io::Error,
   |     ^^^^^^^^^
